bika/lims/browser/analyses/view.py

```diff
@@ -261,6 +261,10 @@ def is_analysis_edition_allowed(self, analysis_brain):
             # lab analyses.
             return False
 
+        # Check if the user is allowed to enter a value to to Result field
+        elif not self.has_permission("Field: Edit Result", analysis_obj):
+            return False
```

Comment:
This condition (L#265) will only be met if:

1. The point of capture of the analysis is "lab"
2. "EditResults" is granted for the current user and analysis
3. The user does not have permission "Field: Edit Result" granted for this object.

This is correct, but since this is an `elif`, the following might happen:

1. The analysis' point of capture is "field" L#253
2. The user has the permission "EditFieldResults" granted L#255

In this scenario, the condition from L#265 is not evaluated (because is an `elif`) and hence, the permission "Field: Edit Result" is never checked. As a result, a "Sampler" ("EditFieldResults" permission granted) will be able to set a value for field Result, even if he/she does not have the permission "Field: Edit Result" granted (or at least, the field will be displayed as editable). At a first glance, seems that converting this `elif` to an `if` would be enough to make this work, cause then, the permission "Field: Edit Result" will always be evaluated, regardless of the point of capture.

Not sure if I am missing something